specs/text-field/text-field.md

```diff
@@ -0,0 +1,142 @@
+# Text Field
+
+## Overview
+
+An implementation of a generic input for textual values as a form-connected web-component.
+
+### Use Cases
+
+Used anywhere an author might otherwise use:
+- input[type="text"]
+- input[type="email"]
+- input[type="password"]
+- input[type="tel"]
+- input[type="url"]
+
+### Features
+
+1. Form Association where supported
+2. Focus Delegation
+
+### Risks and Challenges
+
+We want general feature-parity between this component and the typical corresponding inputs. Most of these concerns are addressed by https://github.com/microsoft/fast-dna/pull/2723, but some stateful nuance will need to be handled by this component.
+
+### Prior Art/Examples
+
+- [FAST-DNA (React)](https://explore.fast.design/components/text-field)
+- [Material UI](https://material-ui.com/components/text-fields/)
+- [Lightning Design](https://www.lightningdesignsystem.com/components/input/)
+- [Carbon Design](https://www.carbondesignsystem.com/components/text-input/code)
+- [Ant Design](https://ant.design/components/input/)
+- [Atlassian](https://atlaskit.atlassian.com/packages/core/textfield)
+- [Office Fabric](https://developer.microsoft.com/en-us/fabric#/controls/web/textfield)
+- [Windows (UWP)](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/text-box)
+
+---
+
+### API
+
+Extends form associated custom element
+
+*Component Name*
+- `fast-text-field`
+
+*Attrs* - Adapted from [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/text)
```

Comment:
How many of these are just forwarding to the internal input element? Should I create a decorator similar to `@attr` that handles forwarding to another element inside the template?

Reply:
I would anticipate everything down to appearance would essentially be forwarded to the input. I'm trying to think of other scenarios like this where we would forward a significant amount of attrs down inside the template. I *think* this is a rare case, but I could be wrong.